subprojects/plugin-use/src/main/java/org/gradle/plugin/use/resolve/internal/AlreadyOnClasspathPluginResolver.java

```diff
@@ -34,26 +34,41 @@
     private static final Factory<ClassPath> EMPTY_CLASSPATH_FACTORY = Factories.constant(ClassPath.EMPTY);
 
     private final PluginResolver delegate;
-    private final PluginRegistry corePluginRegistry;
     private final PluginDescriptorLocator pluginDescriptorLocator;
     private final ClassLoaderScope parentLoaderScope;
     private final PluginInspector pluginInspector;
 
-    public AlreadyOnClasspathPluginResolver(PluginResolver delegate, PluginRegistry corePluginRegistry, ClassLoaderScope parentLoaderScope, PluginDescriptorLocator pluginDescriptorLocator, PluginInspector pluginInspector) {
+    public AlreadyOnClasspathPluginResolver(PluginResolver delegate, ClassLoaderScope parentLoaderScope, PluginDescriptorLocator pluginDescriptorLocator, PluginInspector pluginInspector) {
         this.delegate = delegate;
-        this.corePluginRegistry = corePluginRegistry;
         this.pluginDescriptorLocator = pluginDescriptorLocator;
         this.parentLoaderScope = parentLoaderScope;
         this.pluginInspector = pluginInspector;
     }
 
     @Override
     public void resolve(PluginRequestInternal pluginRequest, PluginResolutionResult result) {
+        validatePluginRequest(pluginRequest);
+
         PluginId pluginId = pluginRequest.getId();
-        if (isCorePlugin(pluginId) || !isPresentOnClasspath(pluginId)) {
+        if (hasGradleNamespace(pluginId)) {
```

Comment:
Some of the internal Kotlin plugins (`EmbeddedKotlinPluginTest`) have the `org.gradle` namespace, are on the classpath according to `isPresentOnClasspath(pluginId)`, however fail to resolve with `resolveAlreadyOnClasspath(pluginId, result)`. This clause intercepts them to be resolved using the delegate.

Reply:
⭕️ Maybe `validatePluginRequest(...)` could become something like `parsePluginRequest(...)` and return some value to simplify the work done on this method? It could remove the need to call `hasGradleNamespace` again on this method, for instance.